plugins/by-name/blink-cmp/default.nix

```diff
@@ -61,16 +45,18 @@ lib.nixvim.neovim-plugin.mkNeovimPlugin {
       '';
     };
 
-    documentation = {
-      auto_show = defaultNullOpts.mkBool false ''
-        Enables automatically showing documentation when typing.
-      '';
-      auto_show_delay_ms = defaultNullOpts.mkUnsignedInt 500 ''
-        Delay, in milliseconds, after which documentation is shown.
-      '';
-      update_delay_ms = defaultNullOpts.mkUnsignedInt 50 ''
-        Delay, in milliseconds, after which documentation is updated.
-      '';
+    windows = {
+      documentation = {
+        auto_show = defaultNullOpts.mkBool false ''
+          Enables automatically showing documentation when typing.
+        '';
+        auto_show_delay_ms = defaultNullOpts.mkUnsignedInt 500 ''
+          Delay, in milliseconds, after which documentation is shown.
+        '';
+        update_delay_ms = defaultNullOpts.mkUnsignedInt 50 ''
+          Delay, in milliseconds, after which documentation is updated.
+        '';
+      };
```

Comment:
While not strictly necessary, it'd be nice to have some warnings added when the old attrs are defined in `cfg.settings`...

@khaneliman has been spearheading this technique. Here's an example of a recent deprecation he made (although for several more options than we need here): https://github.com/nix-community/nixvim/blob/aa06b176e78c9ae9e779e605cab61c9d8681a54e/plugins/by-name/rest/default.nix#L380-L428

Reply:
In this scenario a warning would alert users who had followed our docs and defined `settings.documentation` that they should actually be using `settings.window.documentation`.

It could probably be as simple as 
```nix
warnings = lib.optional (cfg.settings ? documentation) ''
  Nixvim(plugins.blink): some warning about using settings.window.documentation instead...
'';
```